### PR TITLE
Move cert generator out of docker, into a stand-alone tarball

### PR DESCRIPTION
**Pull request by mook-as**
Should work with linux + darwin

### CODE REVIEW

File: bin/cert-generator-wrapper.sh.in [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+set -o errexit
+set -o nounset
+
+usage() {
+	cat <<EOF
+$(basename "${0}"): SCF Certificate Generator
+
+  -d <domain>:     Sets the top level domain for the cluster
```
Comment (andreas-kupries) [inline bin/cert-generator-wrapper.sh.in]: Per line 42 below this is required information. IMHO the help should mention that. Or make it a non-option argument.

File: bin/cert-generator-wrapper.sh.in [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  -h:              Displays this help message
+  -n <namespace>:  Sets namespace, default: 'cf'
+  -o <output dir>: Sets output directory, default: \`pwd\`
```
Comment (andreas-kupries) [inline bin/cert-generator-wrapper.sh.in]: Should possibly mention that the script assumes that out_dir exists and will not create it. Consequence of lines 33ff

File: bin/cert-generator-wrapper.sh.in [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  if test "${host_os}" = '@@CERTSTRAP_OS@@' ; then
+    export PATH="${PATH}:${PWD}/scripts"
+  else
+    printf 'certstrap binary not found, and the embedded one is for @@CERTSTRAP_OS@@, not %s\n' "${host_os}" >&2
+    exit 1
+  fi
+fi
+
+echo "env:" > "${out_dir}/uaa-cert-values.yaml"
+"${scipts_dir}/generate-uaa-certs.sh" -e "${env_dir}" >(perl -pe 's@(.+?)=(.+)@  $1: "$2"@' >> "${out_dir}/uaa-cert-values.yaml") >/dev/null
```
Comment (andreas-kupries) [inline bin/cert-generator-wrapper.sh.in]: For us non-perlians, or the unlucky maintainer-to-come, what does this perl code do, and why ?
IOW this is a request for script comments explaining this a bit more.

File: make/cert-generator [hunk 4]
```diff
@@ -3,17 +3,24 @@
 set -o errexit -o nounset -o pipefail
 
 GIT_ROOT=${GIT_ROOT:-$(git rev-parse --show-toplevel)}
-certgen_docker_root=${certgen_docker_root:-$(mktemp -d)}
+certgen_root=${certgen_docker_root:-$(mktemp -d)}
```
Comment (andreas-kupries) [inline make/cert-generator]: Should the 2nd `certgen_docker_root` also be a `certgen_root` ?

Comment (mook-as) [inline make/cert-generator]: Yes, it should.

File: bin/cert-generator-wrapper.sh.in (hunk 1 above)
Comment (mook-as) [inline bin/cert-generator-wrapper.sh.in]: Fixed in 47750710991b849824c4854502438dd2f0e0336b

File: bin/cert-generator-wrapper.sh.in (hunk 2 above)
Comment (mook-as) [inline bin/cert-generator-wrapper.sh.in]: Fixed in 47750710991b849824c4854502438dd2f0e0336b

File: bin/cert-generator-wrapper.sh.in (hunk 3 above)
Comment (mook-as) [inline bin/cert-generator-wrapper.sh.in]: Comment added in 47750710991b849824c4854502438dd2f0e0336b

File: make/cert-generator (hunk 4 above)
Comment (mook-as) [inline make/cert-generator]: Fixed in 47750710991b849824c4854502438dd2f0e0336b

**Pull request merged**